### PR TITLE
Non-quadratic list, set stringify (like map)

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: lib/core/collections.toit [hunk 1]
```diff
@@ -539,12 +539,16 @@ abstract class List extends CollectionBase:
     (to - from).repeat: this[it + from] = block.call it
 
   /** See $super. */
-  stringify -> string:
-    str := "["
-    size.repeat:
-      if it != 0: str = str + ", "
-      str = str + this[it].stringify
-    return str + "]"
+  stringify:
```
Comment (kasperl) [inline lib/core/collections.toit:542]: Mixin for the win?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/core/collections.toit [hunk 2]
```diff
@@ -539,12 +539,16 @@ abstract class List extends CollectionBase:
...
+    key-strings := []
+    size := 0
+    do: | key |
+      key-string := key.stringify
+      size += key-string.size + 2
+      if size > MAX-PRINT-STRING_:
+        return "[$(key-strings.join ", ")..."
```
Comment (kasperl) [inline lib/core/collections.toit:549]:
```suggestion
        return "[$(key-strings.join ", ")...]"
```

File: lib/core/collections.toit [hunk 3]
```diff
@@ -2626,12 +2630,15 @@ class Set extends HashedInsertionOrderedCollection_ implements Collection:
...
-      if first: first = false else: str = str + ", "
-      str = str + it.stringify
-    return str + "}"
+    key-strings := []
+    size := 0
+    do: | key |
+      key-string := key.stringify
+      size += key-string.size + 2
+      if size > MAX-PRINT-STRING_:
+        return "{$(key-strings.join ", ")..."
```
Comment (kasperl) [inline lib/core/collections.toit:2639]:
```suggestion
        return "{$(key-strings.join ", ")...}"
```

File: lib/core/collections.toit (hunk 2 above)
Comment (kasperl) [reply lib/core/collections.toit:549]: I see that this isn't what we do for maps.... Hmm.

File: lib/core/collections.toit (hunk 1 above)
Comment (floitsch) [inline lib/core/collections.toit:542]: Keep `-> string`

And add it below.

**Pull request merged**